src/main/java/org/jabref/logic/importer/fetcher/MathSciNet.java

```diff
@@ -102,34 +100,141 @@ public URL getUrlForIdentifier(String identifier) throws URISyntaxException, Mal
-
             List<BibEntry> entries = new ArrayList<>();
+
             try {
                 // Depending on the type of query we might get either a json object or directly a json array
                 JsonNode node = new JsonNode(response);
+
                 if (node.isArray()) {
                     JSONArray entriesArray = node.getArray();
                     for (int i = 0; i < entriesArray.length(); i++) {
-                        String bibTexFormat = entriesArray.getJSONObject(i).getString("bib");
-                        entries.addAll(bibtexParser.parseEntries(bibTexFormat));
+                        JSONObject entryObject = entriesArray.getJSONObject(i);
+                        BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                        entries.add(bibEntry);
                     }
                 } else {
                     var element = node.getObject();
-                    JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
-                    for (int i = 0; i < entriesArray.length(); i++) {
-                        String bibTexFormat = entriesArray.getJSONObject(i).getString("bibTexFormat");
-                        entries.addAll(bibtexParser.parseEntries(bibTexFormat));
+
+                    if (element.has("all")) {
+                        JSONArray entriesArray = element.getJSONObject("all").getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
+                    } else if (element.has("results")) {
+                        JSONArray entriesArray = element.getJSONArray("results");
+                        for (int i = 0; i < entriesArray.length(); i++) {
+                            JSONObject entryObject = entriesArray.getJSONObject(i);
+                            BibEntry bibEntry = jsonItemToBibEntry(entryObject);
+                            entries.add(bibEntry);
+                        }
                     }
                 }
-            } catch (JSONException | TokenMgrException e) {
+            } catch (JSONException | ParseException e) {
                 LOGGER.error("An error occurred while parsing fetched data", e);
                 throw new ParseException("Error when parsing entry", e);
             }
             return entries;
         };
     }
 
+    private BibEntry jsonItemToBibEntry(JSONObject item) throws ParseException {
+        try {
+            BibEntry entry = new BibEntry();
+            entry.setType(StandardEntryType.Article);
+
+            // Define the field mappings
+            Map<StandardField, String[]> fieldMappings = Map.ofEntries(
+                    Map.entry(StandardField.TITLE, new String[]{"titles", "title"}),
```

Comment:
To be honest, I don't really don't get this, what is the purpose of the String array and why are there some fields duplicated?

Reply:
Hi, following is a snippet from the "results" JSON that we had in the issue page. I've used a [JSON Formatter](https://jsonformatter.org/json-pretty-print) to visualize it better.
![image](https://github.com/JabRef/jabref/assets/74734844/ad38d3cf-49da-4368-aeca-ba97d3c981de)
To retrieve the journal value, we need to navigate through the issue object twice, then the journal object, and finally access the shortTitle value. Here, the String array {"issue", "issue", "journal", "shortTitle"} represents the path to reach the journal value in the JSON response. 
In a nutshell, the String array associated with each StandardField represents the path to reach the desired value in the JSON response, if I get your question correctly.

Reply:
Okay, thanks for the explanation. Was not directly obvious. I will try to think if this can be made a bit better. I try to check back tomorrow.

Reply:
Sure, please let me know!

Reply:
Please add a Java comment to explain.